kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -72,7 +74,8 @@ public SnapshotImpl(
       Protocol protocol,
       Metadata metadata,
       Committer committer,
-      SnapshotQueryContext snapshotContext) {
+      SnapshotQueryContext snapshotContext,
+      Optional<Long> inCommitTimestampOpt) {
```

Comment:
Please clarify when this is present/not. If it's not present, does it mean this snapshot does not have ICT enabled and does not have an inCommitTimestamp? Or does it mean it just hasn't been read?

I expected the `getTimestamp` code to have been updated with this and was a little surprised it hadn't been.

Reply:
Dah. Fair point. We keep running into this issue again and again:
- there is some optional value
- someone might _know_ that optional value
- we have to inject that informormation into Kernel
- Optional.empty could mean -- I *know* the value is empty, or it could mean I don't know the value